Added slides on godbolt and cppinsights (#232)
hsf-training/cpluspluscourse: talk/tools/webtools.tex

\subsection[web]{Web tools}

\begin{frame}
  \frametitle{Godbolt / Compiler Explorer }
  \begin{block}{Concept}
    An online generic compiler with immediate feedback.

>     Allows to :
>     \begin{itemize}
>     \item compile online any code against any version of any compiler
>     \item inspect the assembly generated
>     \item use external libraries (over 50 available !)
>     \item run the code produced
>     \item use tools, e.g. ldd, include-what-you-use, ...
>     \item share small pieces of code via permanent short links

Reviewer: I wouldn't use "Allow to", see e.g. [english stackexchange](https://english.stackexchange.com/questions/60271/grammatical-complements-for-allow/60285#60285).

We could use:
```suggestion
    Allows:
    \begin{itemize}
    \item compiling any code online against any version of any compiler
    \item inspecting the assembly generated
    \item use of external libraries (over 50 available !)
    \item running the code produced
    \item using tools, e.g.\ ldd, include-what-you-use, ...
    \item sharing small pieces of code via permanent short links
```

    \end{itemize}
  \end{block}
  \begin{exampleblock}{Typical usage}
    \begin{itemize}
    \item check small pieces of code on different compilers

>     \item check some new \cpp functionnality and its support

Reviewer 2: ```suggestion
    \item check some new \cpp functionality and its support
```

Author: done

    \item optimize small piece of code
    \item NOT relevant for large codes
    \end{itemize}
  \end{exampleblock}
\end{frame}

\begin{frame}
  \frametitle{Godbolt by example}
  \begin{block}{Check effect of optimization flags}
    \url{https://godbolt.org/z/Pb8WsWjEx}
    \begin{itemize}
    \item Check generated code with -O0, -O1, -O2, -O3
    \item See how it gets shorter and simpler
    \end{itemize}
  \end{block}
  \includegraphics[width=\textwidth]{tools/godbolt.jpg}
\end{frame}

\begin{frame}
  \frametitle{cppinsights}
  \begin{block}{Concept}
    Reveals the actual code behind some \cpp syntactic features
    \begin{itemize}
    \item lambdas
    \item range based loops
    \item templates
    \item initializations
    \item auto
    \item ...
    \end{itemize}
  \end{block}
  \begin{exampleblock}{Typical usage}
    \begin{itemize}

>     \item understand how things are in the back of \cpp syntax

Reviewer 2: ```suggestion
    \item understand how things work behind the \cpp syntax
```

    \item debug some non working pieces of code
    \end{itemize}
  \end{exampleblock}
\end{frame}

\begin{frame}
  \frametitle{cppinsights by example}

>   \begin{block}{Check how range based loop work}

Reviewer 2: ```suggestion
  \begin{block}{Check how range-based loops work}
```

    \url{https://cppinsights.io/s/b886aa76}
    \begin{itemize}

>     \item See how they map to regular iterator

Reviewer 2: ```suggestion
    \item See how they map to regular iterators
```
